lib/xebow.ex

```diff
@@ -100,33 +121,66 @@ defmodule Xebow do
   # Server Implementations:
 
   @impl GenServer
-  def init(types) do
-    count_of_active_animations = length(types)
-
-    active_animations =
-      types
-      |> Stream.map(&initialize_animation/1)
-      |> Stream.with_index()
-      |> Stream.map(fn {animation, index} -> {index, animation} end)
-      |> Enum.into(%{})
+  def init(_) do
+    active_animation_types =
+      if Settings.active_animations_file_exists?() do
+        case Settings.load_active_animations() do
+          {:ok, active_animation_types} ->
+            active_animation_types
+
+          {:error, reason} ->
+            Logger.warn("Failed to load active animations: #{inspect(reason)}")
+            Animation.types()
+        end
+      else
+        Animation.types()
+      end
```

Comment:
Updated in c715fed.

Good thought. That led me to another thought that removes unnecessary code...

I realized the `Settings.active_animations_file_exists?` guard isn't necessary, since the file should be created on application start. In the rare event this file is deleted after the app starts but before the `Xebow` GenServer starts, `load_active_animations` will return `{:error, :enoent}`, which will be handled by the `case` and everything will be fine. The error case is also the same result as was in the `else` block, so this deduplicates that code.

Reply:
Awesome, that makes a lot of things simpler. :tada: